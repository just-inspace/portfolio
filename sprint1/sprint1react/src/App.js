import React, { Component } from "react";
import { Route, HashRouter } from "react-router-dom";

import SideNav from "./components/sideNav";
import Home from "./components/home";
import Projects from "./components/projects";
import Resume from "./components/resume";
import Contact from "./components/contact";

class App extends Component {
	render() {
		return (
			<HashRouter>
				<div id="app">
					<SideNav />
					<div id="page">
						<Route exact path="/" component={Home} />
						<Route path="/projects" component={Projects} />
						<Route path="/resume" component={Resume} />
						<Route path="/contact" component={Contact} />
					</div>
				</div>
			</HashRouter>
		);
	}
}

export default App;
