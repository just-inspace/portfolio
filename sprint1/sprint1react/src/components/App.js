import React, { Component } from "react";
import SidebarContainer from "./sidebarContainer";
import PageContainer from "./pageContainer";

class App extends Component {
	render() {
		return (
			<div id="app">
				<SidebarContainer />
				<PageContainer />
			</div>
		);
	}
}

export default App;
