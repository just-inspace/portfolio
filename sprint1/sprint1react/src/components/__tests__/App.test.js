import { shallow } from "enzyme";
import App from "../App.js";
import React from "react";

describe("App", () => {
	it("renders correctly", () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});
});

it("renders App with SidebarContainer", () => {
	const wrapper = shallow(<App />);

	expect(wrapper.find(SidebarContainer).length).toEqual(1);
});

it("renders App with PageRouter", () => {
	const wrapper = shallow(<App />);

	expect(wrapper.find(PageContainer).length).toEqual(1);
});
