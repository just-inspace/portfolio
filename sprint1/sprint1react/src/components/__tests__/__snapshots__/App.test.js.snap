// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        id="app"
      >
        <SideNav />
        <div
          id="page"
        >
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/projects"
          />
          <Route
            component={[Function]}
            path="/resume"
          />
          <Route
            component={[Function]}
            path="/contact"
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideNav />,
          <div
            id="page"
          >
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/projects"
            />
            <Route
              component={[Function]}
              path="/resume"
            />
            <Route
              component={[Function]}
              path="/contact"
            />
          </div>,
        ],
        "id": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/projects"
              />,
              <Route
                component={[Function]}
                path="/resume"
              />,
              <Route
                component={[Function]}
                path="/contact"
              />,
            ],
            "id": "page",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/projects",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/resume",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/contact",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          id="app"
        >
          <SideNav />
          <div
            id="page"
          >
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/projects"
            />
            <Route
              component={[Function]}
              path="/resume"
            />
            <Route
              component={[Function]}
              path="/contact"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SideNav />,
            <div
              id="page"
            >
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/projects"
              />
              <Route
                component={[Function]}
                path="/resume"
              />
              <Route
                component={[Function]}
                path="/contact"
              />
            </div>,
          ],
          "id": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  path="/projects"
                />,
                <Route
                  component={[Function]}
                  path="/resume"
                />,
                <Route
                  component={[Function]}
                  path="/contact"
                />,
              ],
              "id": "page",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/projects",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/resume",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/contact",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
